wasmi-impl/src/lib.rs

```diff
@@ -1,10 +1,21 @@
 #![no_std]
+extern crate alloc;
+
+use core::cmp;
+use alloc::vec::{self, Vec};
 use wasmi::{
-    self, memory_units::Pages, ExternVal, ImportsBuilder, MemoryInstance, ModuleInstance,
-    NopExternals, RuntimeValue,
+    self, memory_units::Pages, ExternVal, ImportsBuilder, MemoryInstance,
+    ModuleInstance, NopExternals, RuntimeValue,
 };
 
 static ENTRYPOINT: &str = "exec";
+static MAX_MEMORY_KIB: usize = 128 * 1024;
+static MAX_PAGES: usize = match MAX_MEMORY_KIB % 64 {
+    0 => MAX_MEMORY_KIB / 64,
+    _ => MAX_MEMORY_KIB / 64 + 1,
+};
+static MEM_SMALL_BYTES: usize = 128 / 8;
+static PAGE_SIZE_BYTES: usize = 64 * 1024;
```

Comment:
We should probably use `wasmi` constants here, where possible?

https://docs.rs/wasmi/latest/wasmi/index.html#constants

Reply:
(We should also probably prefer `const`, if `static` isn't necessary.)

Reply:
Changed all but one to `const`.  Need confirmation from @longtomjr whether or not the `ENTRYPOINT` strictly needs to be a `static`